sycl/include/sycl/sycl_span.hpp

```diff
@@ -193,6 +253,12 @@ struct __is_span_compatible_container<
                                   _ElementType (*)[]>,
             std::nullptr_t>::type>> : public std::true_type {};
 
+template <class _It>
+using __is_contiguous_iterator = std::bool_constant<
+    std::is_pointer_v<_It> ||
+    std::is_same_v<typename std::iterator_traits<_It>::iterator_category,
+                   std::random_access_iterator_tag>>;
```

Comment:
Is that per spec? I can easily create non-contiguous iterator with random accesses (e.g. view of elements with even indices).